ocaml/xapi/create_misc.ml

```diff
@@ -687,105 +673,81 @@ let create_host_cpu ~__context host_info =
 let create_pool_cpuinfo ~__context =
   let open Map_check in
   let open Cpuid_helpers in
+  let open Xapi_xenops_queue in
+  let open Xenops_interface in
+  let module Xenopsd = (val make_client (default_xenopsd ()) : XENOPS) in
+  let dbg = Context.string_of_task __context in
   let all_host_cpus =
     List.map
       (fun (r, s) -> (r, s.API.host_cpu_info))
       (Db.Host.get_all_records ~__context)
   in
-  let getfdefault ~defaultf key host =
-    let default = getf defaultf host in
-    getf ~default key host
-  in
   let merge pool (hostref, host) =
+    let combine a b =
+      match (CPU_policy.to_string a, CPU_policy.to_string b) with
```

Comment:
This could have an 'is_empty' to avoid converting to string only to check for emptyness (well it is a string currently, so it is probably fine as it is for now)?